apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -233,6 +234,20 @@ private void DeleteSessionFireAndForget(PooledSession session)
             ConsumeBackgroundTask(DeleteSessionAsync(session), "session deletion");
         }
 
+        /// <summary>
+        /// Rolls back the specified transaction, if any, in a fire-and-forget manner.
+        /// </summary>
+        /// <param name="session">The name of the session that owns the transaction.</param>
+        /// <param name="transactionID">The ID of the transaction to rollback. May be null, in which case this method is a no-op.</param>
+        private void MaybeRollbackTransactionFireAndForget(SessionName session, ByteString transactionID)
```

Comment:
Here's where I would pass the whole PooledSession, check whether we need to rollback and do it.